src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx

```diff
@@ -37,60 +38,52 @@ export class DictionaryConcepts extends Component {
     loading: PropTypes.bool.isRequired,
     filterBySource: PropTypes.func.isRequired,
     filterByClass: PropTypes.func.isRequired,
+    paginateConcepts: PropTypes.func.isRequired,
+    totalConceptCount: PropTypes.number.isRequired,
   };
 
   constructor(props) {
     super(props);
     this.state = {
-      conceptsCount: getTotal(),
+      conceptsCount: this.props.totalConceptCount,
       searchInput: '',
+      conceptLimit: 10,
+      conceptOffset: 0,
     };
     autoBind(this);
   }
 
   componentDidMount() {
     this.fetchConcepts();
+  }
+
+  componentWillReceiveProps(nextProps) {
     const {
       match: {
         params: { collectionName, type, typeName },
       },
-    } = this.props;
+    } = nextProps;
     localStorage.setItem('dictionaryId', this.props.match.params.collectionName);
     localStorage.setItem('type', this.props.match.params.type);
     localStorage.setItem('typeName', this.props.match.params.typeName);
-    this.setState({ collectionName, type, typeName });
-  }
-
-  componentWillReceiveProps(nextProps) {
-    if (this.state.searchInput) {
-      this.setState({
-        conceptsCount: nextProps.concepts.length,
-      });
-    } else {
-      this.setState({
-        conceptsCount: getTotal(),
-      });
-    }
+    this.setState({
+      collectionName,
+      type,
+      typeName,
+      conceptsCount: nextProps.totalConceptCount,
+    });
   }
 
-  fetchConcepts(query = '', limit = 10, filterParams = null, filterName = null) {
```

Comment:
Does the change from limit 10 to 0 mean you are paginating from client instead of server?

Reply:
yes, it's faster.

Reply:
I also needed to get the total number of concepts at every point in time whether I'm searching or filtering.

Reply:
What happens when a server has too many concepts? Do you still want to load all of them to the client?

Reply:
yeah, for now, that's the only way to get the total number of concepts while filtering by sources or class.

Reply:
Let me ask it this way. What was the 10 originally accomplishing?

Reply:
initially, I was fetching just 10 concepts whether a user was searching or filtering.

Reply:
That limit parameter was intentionally put in place for that. So ignoring it may result into running out of server memory when the concepts are very many. Not so?

Reply:
I think the server can handle it. You may refer to - https://github.com/OpenConceptLab/oclapi/wiki#pagination-and-limiting-the-number-of-returned-results

Reply:
Let me put it this way, why do you think they put a provision for the limit parameter?

Reply:
I understand what you are saying, my point is I need to get the total number of concepts each time I fetch the concepts from the server, right now it will not be possible if I use the limit parameter.

Reply:
So with your change, are you simply getting the total number of concepts? Or are you actually getting the concepts?

Reply:
I'm getting the concepts. There's currently no way to get only the total number of concepts when filtering or searching.

Reply:
So my point is, the original code was safer because it does not load all the concepts. And this was achieved by the use of the limit parameter. Getting rid of this parameter looks like a design mistake unless you can convince me otherwise.

Reply:
The initial code was a quick hack, I only got the total number of concepts in a dictionary because I saved it in the localStorage. The issues with the initial code are as follows: 
- when a filter returns eleven concepts, the user will only see ten with no idea there could be more because there's no way to know if there's actually more.
- The filter sidebar cannot be populated with items that are not available. i.e - if the first ten concepts are from the same source or class, the sidebar will be populated by only that type of source. This will clearly defeat the aim of having the sidebar. 
- There are other reasons why I have to fetch everything first, but I will stop here. Refer to the images below for more clarity about my second point.
<img width="1440" alt="with pagination" src="https://user-images.githubusercontent.com/5228540/42817485-291b2610-89c6-11e8-96ca-54b1a78b6a44.png">
<img width="1440" alt="without pagination" src="https://user-images.githubusercontent.com/5228540/42817487-2946b37a-89c6-11e8-803b-f7545a6a7af5.png">